integration_test/ops_agent_test.go

```diff
@@ -1754,6 +1771,51 @@ func TestWindowsEventLogV2(t *testing.T) {
 			t.Fatal(err)
 		}
 
+		// For r3 (the XML logs), verify the following:
+		// - that jsonPayload only has the fields we expect (Message, StringInserts, raw_xml)
+		// - that jsonPayload.raw_xml contains a valid XML document
+		// - that a few sample fields are present in that XML document
+		for _, payload := range payloads["r3"] {
+			log, err := gce.QueryLog(ctx, logger.ToMainLog(), vm, "r3", time.Hour, logMessageQueryForPlatform(vm.Platform, payload), gce.QueryMaxAttempts)
+			if err != nil {
+				t.Fatal(err)
+			}
+			// We don't know (and don't care about) the runtime type graph of log.Payload, so normalize it into a simple map
+			rawJson, err := json.Marshal(log.Payload)
+			if err != nil {
+				t.Fatal(err)
+			}
+			jsonMap := map[string]any{}
+			err = json.Unmarshal(rawJson, &jsonMap)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if len(jsonMap) != 3 ||
+				!hasKeyWithValueType[string](jsonMap, "Message") ||
+				!hasKeyWithValueType[[]any](jsonMap, "StringInserts") ||
+				!hasKeyWithValueType[string](jsonMap, "raw_xml") {
+				t.Fatalf("expected exactly 3 fields in jsonPayload (Message, StringInserts, raw_xml): jsonPayload=%+v", jsonMap)
+			}
+			rawXml := jsonMap["raw_xml"].(string)
+			xmlStruct := struct {
+				System struct {
+					TimeCreated struct {
+						SystemTime string `xml:",attr"`
+					}
+				}
+				EventData struct {
+					Data string
+				}
+			}{}
+			err = xml.Unmarshal([]byte(rawXml), &xmlStruct)
+			if err != nil {
+				t.Fatalf("expected raw_xml to contain a valid XML document: raw_xml=%s, err=%v", rawXml, err)
+			}
+			if len(xmlStruct.EventData.Data) == 0 || len(xmlStruct.System.TimeCreated.SystemTime) == 0 {
```

Comment:
Just curious, why check the length and not:
```go
if xmlStruct.EventData.Data == "" || xmlStruct.System.TimeCreated.SystemTime == "" {
```

Reply:
Wires got crossed